internal/mesh/internal/controllers/gatewayproxy/controller.go

```diff
@@ -32,6 +34,8 @@ func Controller(cache *cache.Cache, trustDomainFetcher sidecarproxy.TrustDomainF
 	return controller.NewController(ControllerName, pbmesh.ProxyStateTemplateType).
 		WithWatch(pbcatalog.WorkloadType, dependency.ReplaceType(pbmesh.ProxyStateTemplateType)).
 		WithWatch(pbmesh.ComputedProxyConfigurationType, dependency.ReplaceType(pbmesh.ProxyStateTemplateType)).
+		WithWatch(pbcatalog.ServiceType, mapper.AllMeshGatewayWorkloadsInPartition).
```

Comment:
I don't think it's sufficient to only reconcile mesh gateways in the same partition as the service that changed.

This is because the service that changed _could_ be a mesh gateway. If the mesh gateway mode is local, then the mesh gateway in the other partition will need to reconcile when this happens.

Reply:
@NiniOak I think it's valuable to have the `ComputedExportedServices` watch as-is. I'd probably boot this `Service` watch to a followup PR since it will have slightly more complex logic